openml/tasks/functions.py

```diff
@@ -236,7 +236,7 @@ def get_task(task_id):
 
     # TODO look into either adding the class labels to task xml, or other
     # way of reading it.
-    class_labels = dataset._retrieve_class_labels()
+    class_labels = dataset._retrieve_class_labels(task.target_feature)
```

Comment:
Okay, I'll make the function public. Also, target_name sounds good to me, although the name comes from OpenML.

This is classification specific, and I should add a comment for that. It's actually necessary when creating the arff file with the results, which needs the real class name (not just the index as given by y) to upload the probabilities for each class to the server.